pennylane/ops/qubit/observables.py

```diff
@@ -181,6 +181,8 @@ class SparseHamiltonian(Observable):
     grad_method = None
 
     def __init__(self, H, wires=None, do_queue=True, id=None):
+        if not isinstance(H, coo_matrix):
+            raise TypeError("Observable must be a scipy sparse coo_matrix.")
```

Comment:
I moved this from the matrix methods, because it is an input check that should be performed upon initialisation, if at all!